packages/editor/src/components/media-placeholder/index.js

```diff
@@ -258,10 +258,10 @@ export class MediaPlaceholder extends Component {
 }
 
 const applyWithSelect = withSelect( ( select ) => {
-	const { hasUploadPermissions } = select( 'core' );
+	const { canUser } = select( 'core' );
 
 	return {
-		hasUploadPermissions: hasUploadPermissions(),
+		hasUploadPermissions: canUser( 'create', 'media', undefined, true ),
```

Comment:
This is a personal preference, but I'm not such a big fan of having to pass undefined as an argument here. Some alternatives (but don't feel like it's something that needs to be changed):
```
defaultTo( canUser( 'create', 'media' ), true );
```

```
canUser( [ 'create', 'media' ], true );
```

```
canUser( { action: 'create', resource: 'media', default: true } );
```